pkg/utils/utils.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+// GetClusterDomainName returns cluster's domain name or error
+func GetClusterDomainName() (string, error) {
+	_, err := os.Stat("/etc/resolv.conf")
+	if err != nil {
+		return "", err
+	}
+	f, err := os.Open("/etc/resolv.conf")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	return getClusterDomainName(f)
+}
+
+func getClusterDomainName(r io.Reader) (string, error) {
+	scanner := bufio.NewScanner(r)
```

Comment:
No strong preference here, but if `/etc/resolv.conf` is larger than 20KB, you probably have other problems.

As such, it's probably a break-even betwene `bufio.NewScanner` and `ioutil.ReadAll`.

Reply:
(Where using ioutil.ReadAll would mean that this method would take a `string` rather than an `io.Reader`.)

Reply:
this file is controlled by kubernetes cluster and mounted by kubelet into the pod, so there is a good chance it has been validated against some obvious issues like excessive size. If you do not mind I would prefer to leave this version.